File: OOAD_HW1/Q4/hippo.java
package Q4;

public class hippo extends pachyderm {

	// Hippo's name
	private String name;
	
	// Default Constructor, don't use.
	public hippo() {
		super("None");
		this.name = "None";
	}
	
	// Constructor for setting the hippo's name, and it's super-class' name.
	public hippo(String name) {
		super(name);
		this.name = name;
	}
}

File: OOAD_HW2/Part B/Hippo.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <iomanip>

using namespace std;

class Hippo: public Pachyderm
{
    protected:
        string name;
        string type;
        //NoiseBehavior *noise;

    public:
        Hippo(string aName, string aType);
};

// Behavior handled by interface via super class.
Hippo::Hippo(string aName, string aType): Pachyderm(aName, aType)
{
    name = aName;
    type = aType;
}



